client/src/main/java/org/eclipse/hono/client/CommandResponse.java

```diff
@@ -96,15 +101,16 @@ public static CommandResponse from(
      * @return The command response or {@code null} if any of correlationId, replyTo and status is null or if the
      *         status code is &lt; 200 or &gt;= 600.
      */
-    public static CommandResponse from(final Buffer payload, final String contentType, final Integer status,
+    public static CommandResponse from(final String tenantId, final String deviceId, final Buffer payload,
```

Comment:
this seems to be the factory method to be used by the AMQP adapter, right? What about simply passing in the `Message` instead of all the params?

Reply:
yes. I think its a good idea. It makes the Adapter code more readable and it can be reused by the adapters.

Reply:
@sophokles73 I have refactored to accept message as a parameter and pushed my changes.